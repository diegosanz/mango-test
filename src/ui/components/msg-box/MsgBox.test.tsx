import React from "react";
import { render } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import MsgBox from "./MsgBox";

test("renders empty error MsgBox", () => {
  render(<MsgBox type="error" />);
});

test("renders empty info MsgBox", () => {
  render(<MsgBox type="info" />);
});

test("renders full error MsgBox", () => {
  render(<MsgBox type="error">Test</MsgBox>);
});

test("renders full info MsgBox", () => {
  render(<MsgBox type="info">Test</MsgBox>);
});

test("snapshot empty error match", () => {
  const tree = TestRenderer.create(<MsgBox type="error" />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("snapshot empty info match", () => {
  const tree = TestRenderer.create(<MsgBox type="info" />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("snapshot full error match", () => {
  const tree = TestRenderer.create(<MsgBox type="error">Test</MsgBox>).toJSON();
  expect(tree).toMatchSnapshot();
});

test("snapshot full error match", () => {
  const tree = TestRenderer.create(<MsgBox type="info">Test</MsgBox>).toJSON();
  expect(tree).toMatchSnapshot();
});
