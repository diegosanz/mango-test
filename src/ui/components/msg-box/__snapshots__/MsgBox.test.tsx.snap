// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot empty error match 1`] = `null`;

exports[`snapshot empty info match 1`] = `null`;

exports[`snapshot full error match 1`] = `
<div
  className="sc-bdfBwQ cjQMhH m-error"
>
  Test
</div>
`;

exports[`snapshot full error match 2`] = `
<div
  className="sc-bdfBwQ cjQMhH m-info"
>
  Test
</div>
`;
